models/synthetic_data/synthetic_data_for_pipeline_transform/generate.py

```diff
@@ -0,0 +1,287 @@
+"""This module will generate data that can be used for pre-training the model used for the transform phase of the
+training data pipeline. """
+# Ideas from Jeannine
+# no lines, colored construction paper, inverse color intensity, paragraph structure of ink, maybe rotated paper,
+# writing on the bottom that curves up because they ran out of space. Also writing may take a shape like a diamond, or
+# an animal etc. Maybe BOLD writing needs to be represented. intersperse art work in the page? Glued on shapes?
+# News clipping cut outs.. can a GAN create samples? Eizibeth did something similar?
+
+# Elizibeth created handwriting based custom fonts..
+
+import random
+import sys
+import uuid
+import numpy
+import numpy as np
+import cv2
+import json
+from multiprocessing import Process
+import os
+import os.path
+import typer
+import time
+from os import path
+import hashlib
+
+# expected input size of model
+input_size = (256 - 64, 256)
+# list of photograph sizes found in current data
+photo_sizes = [(2016, 1512, 3), (4032, 3024, 3), (640, 480, 3)]
+
+
+def create_fake_paper_with_writing(pix_per_mm: float, lines: int = 25, red: [] = [255, 164, 164],
+                                   blue: [] = [128, 128, 255],
+                                   ink: [] = [32, 32, 96], noise: float = 1):
+    """Creates an image that is much like written words on ruled paper, also returns a dict of important information"""
+
+    def get_rnd_col(col=[128, 128, 128], c_noise: int = 128):
+        for i in range(len(col)):
+            if (c_noise / 2) + col[i] > 255:
+                col[i] = 255 - (c_noise / 2)
+
+            if (c_noise / 2) + col[i] < 0:
+                col[i] = (c_noise / 2)
+
+        for i in range(len(col)):
```

Comment:
is this check valid if the previous block eliminates all the vlaues outside the boundary of [0,255]

Reply:
100% true. Removing.